rasterio/_io.pyx

```diff
@@ -660,7 +698,7 @@ cdef class DatasetReaderBase(DatasetBase):
         warnings.warn(
             "read_mask() is deprecated and will be removed by Rasterio 1.0. "
             "Please use read_masks() instead.",
-            FutureWarning,
+            DeprecationWarning,
```

Comment:
My point was that the `Deprecation` warning gets filtered out and isn't visible to users by default, so you need to update the filter to force it to show...